### PR TITLE
Add protection against invalid status codes

### PR DESCRIPTION
**Pull request by gabsn**
- [x] **tested on staging**

### CODE REVIEW

**Comment by talwai** [comment]
apply the same comments from here please https://github.com/DataDog/dd-go/pull/3294

File: model/normalizer.go [hunk 1]
```diff
@@ -152,6 +153,13 @@ func (s *Span) Normalize() error {
 		s.Meta["env"] = NormalizeTag(env)
 	}
 
+	// Status Code
+	if sc, ok := s.Meta["http.status_code"]; ok {
+		if !isValidStatusCode(sc) {
+			delete(s.Meta, "http.status_code")
```
Comment (talwai) [inline model/normalizer.go]: add a `log.Debug` line just so we tell the user we're dropping their meta.

**Comment by talwai** [comment]
quick comment. Looks good otherwise

File: model/normalizer.go (hunk 1 above)
Comment (gabsn) [inline model/normalizer.go]: if we do that, is will quickly become unreadable, is it ok?

Comment (talwai) [inline model/normalizer.go]: use `DEBUG` level so it will not spam by default. and if we turn it on it quickly makes the problem obvious

**Pull request merged**